pages/data-providers/data-provider.jsx

```diff
@@ -1,7 +1,125 @@
+//       Find a better place for it.
+const useDiscoveryStore = (defaultDataProviderUrl) => {
+  const store = useLocalStore(() => ({
+    query: defaultDataProviderUrl,
+    request: null,
+    loading: false,
+    created: null,
+    duplicated: [],
+    error: null,
+  }))
+
+  useDebouncedEffect(
+    () => {
+      if (store.query === '') return
+
+      store.isLoading = true
+
+      // TODO: Call API Here
+      clearTimeout(store.request)
+      store.request = setTimeout(() => {
+        // reset store
+        store.request = null
+        store.created = null
+        store.duplicated = []
+        store.error = null
+
+        if (store.query === 'success.url') store.created = { url: store.query }
+        else if (store.query === 'duplicate.url')
+          store.duplicated = [{ url: store.query }]
+        else if (store.query === 'multiple-duplicates.url')
+          store.duplicated = [{ url: store.query }, { url: store.query }]
+        else store.error = true
+
+        store.isLoading = false
+      }, 1000)
+    },
+    [store.query],
+    500
+  )
+
+  useEffect(() => {
+    store.query = defaultDataProviderUrl
+  }, [defaultDataProviderUrl])
+
+  return store
+}
+
+const getHelperMessage = ({ query, created, duplicated, error }) => {
+  if (error) {
+    return {
+      message:
+        'We could not find any OAI-PMH endpoint.' +
+        ' Please try it with another URL.',
+      variant: 'error',
+    }
+  }
+
+  if (created) {
+    return {
+      message: `We found OAI-PMH endpoint at ${created.url}`,
+      variant: 'success',
+    }
+  }
+
+  if (duplicated.length) {
+    return {
+      message:
+        `${query} ${
+          duplicated.length > 1 ? `and ${duplicated.length - 1} more are` : 'is'
+        } already` +
+        ' in our system. If you host more than one repository on the same domain,' +
+        ' please specify exact OAI-PMH endpoint',
+      variant: 'error',
+    }
+  }
+
+  return {
+    message:
+      'It can be any URL: homepage address, any data resource address,' +
+      ' OAI-PMH endpoint or RIOXX endpoint',
+    variant: 'normal',
+  }
+}
+
+const DataProviderPage = () => {
+  const router = useRouter()
+  const store = useDiscoveryStore(router.query?.dataProviderUrl || '')
+  const handleSubmitForm = () => {
+    // TODO: Handle next step. Show success message
+    // router.push('/next-step')
+  }
+
+  const { message, variant } = getHelperMessage(store)
+
+  return useObserver(() => (
+    <DataProviderPageTemplate
+      url={store.query}
+      // TODO: Discuss URL validation.
+      //       Maybe we shouldn't be validating it on frontend because there
+      //       is a lot of cases and type="url" is not sufficient.
+      //       For example google.com is validated as a wrong input and
+      //       that's not quite true. Maybe we should keep the validation
+      //       on backend entirely.
+      onUrlChange={(event) => {
```

Comment:
If you pass `onSubmit` keep it consistent with form event listeners, i.e. call it `onChange`

Reply:
The thing is that this is covering only input change and not the whole form hence `url` there. I couldn't use form change event because React was complaining about uncontrolled input. If you put value to input you need to have onChange handler there too.

Reply:
Can this be faked? Like you pass `onChange` here which is passed directly to the input. It can be tracked using `event.target.name` anyway.